tensorflow_datasets/image/lsun.py

```diff
@@ -156,6 +157,9 @@ def _generate_examples(self, extracted_dir, file_path):
     with tf.Graph().as_default():
       path = os.path.join(extracted_dir, file_path, "data.mdb")
       dataset = _make_lmdb_dataset(path)
-      for i, (_, jpeg_image) in enumerate(tfds.as_numpy(dataset)):
-        record = {"image": io.BytesIO(jpeg_image)}
+      for i, (id_bytes, jpeg_image) in enumerate(tfds.as_numpy(dataset)):
+        record = {
+          "id": id_bytes.decode("utf-8"),
```

Comment:
I'm not really sure about changing the structure of the dataset by adding in a new `id` feature as it might cause some issues for the previous users. 

Also, since the different configs consist of images belonging to the same category (of each config), I don't think the `id` feature would be required. Let me know your thoughts on this.

Reply:
Why do you think adding a feature would break an existing use case? Wouldn't an existing user just ignore the presence of the additional feature?

For my own use case, I need to have the `id` feature, because I want to be able to perform certain per-image analysis tasks -- basically, I want to be able to associate arbitrary additional metadata with specific images, and having a static per-image identifier makes that much easier.

Reply:
Since, the dataset does not currently have supervised keys (which would have returned a tuple, instead of a dict ), I think you are right that the new feature would only add as an additional for the existing user base.

In that case, I think it would be good to add the supervised keys now (as `('id, 'image')`) as shown [here](https://www.tensorflow.org/datasets/add_dataset#_info_dataset_metadata).

I'll send a PR to update the required documentation once this PR gets approved from a maintainer or merged. Thanks!